src/main/java/com/minecolonies/coremod/commands/ActionArgumentType.java

```diff
@@ -0,0 +1,360 @@
+import com.minecolonies.coremod.colony.ColonyManager;
+import com.minecolonies.coremod.colony.managers.ICitizenManager;
+import com.minecolonies.coremod.commands.CommandEntryPointNew.ActionMenuHolder;
+import com.mojang.authlib.GameProfile;
+
+import net.minecraft.command.ICommandSender;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.math.BlockPos;
+import net.minecraftforge.common.util.FakePlayer;
+
+public enum ActionArgumentType
+{
+    PLAYER("player-expression", 0),
+    COLONY("colony-id", 0),
+    CITIZEN("citizen-id or full-name", 2),
+    COORDINATE_X("x-coordinate", 0),
+    COORDINATE_Y("y-coordinate", 0),
+    COORDINATE_Z("z-coordinate", 0),
+    BOOLEAN("boolean", 0),
+    INTEGER("integer", 0),
+    ;
+
+    public enum Is
+    {
+        REQUIRED,
+        OPTIONAL
+    }
+
+    @Nonnull private final String usageValue;
+    private final int allowedSpaceCount;
+
+    ActionArgumentType(@Nonnull final String usageValue, final int allowedSpaceCount)
+    {
+        this.usageValue = usageValue;
+        this.allowedSpaceCount = allowedSpaceCount;
+    }
+
+    public String getUsageValue()
+    {
+        return usageValue;
+    }
+
+    public int allowedSpaceCount()
+    {
+        return allowedSpaceCount;
+    }
+
+    private List<String> getOnlinePlayerNames(@NotNull final MinecraftServer server)
+    {
+        final String[] onlinePlayerNames = server.getOnlinePlayerNames();
+        return Arrays.asList(onlinePlayerNames);
+    }
+
+    private List<String> getColonyIdStrings()
+    {
+        final List<Colony> colonyList = ColonyManager.getColonies();
+        final List<String> colonyIdList = new ArrayList<>(colonyList.size());
+        for (final Colony colony : colonyList)
+        {
+            colonyIdList.add(String.valueOf(colony.getID()));
+        }
+        return colonyIdList;
+    }
+
+    private List<String> getCitizenNames()
+    {
+        final List<Colony> colonyList = ColonyManager.getColonies();
+        final List<String> citizenNameList = new ArrayList<>();
+        for (final Colony colony : colonyList)
+        {
+            final List<CitizenData> citizenDataList = colony.getCitizenManager().getCitizens();
+            for (final CitizenData citizenData : citizenDataList)
+            {
+                citizenNameList.add(citizenData.getName());
+            }
+        }
+        return citizenNameList;
+    }
+
+    public List<String> getTabCompletions(@NotNull final MinecraftServer server,
+            @Nullable final BlockPos pos,
+            final String potentialArgumentValue)
+    {
+        switch (this)
+        {
+            case INTEGER:
+                return Collections.emptyList();
+            case BOOLEAN:
+                return Arrays.asList(new String[] {"true", "false"});
+            case COORDINATE_X:
+            case COORDINATE_Y:
+            case COORDINATE_Z:
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this switch case number of lines from 24 to at most 10, for example by extracting code into methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1151)